src/integrations/react.js

```diff
@@ -1,9 +1,50 @@
-import { createElement, Children, Component } from 'react';
+import { createContext, createElement, useContext, useState, useEffect } from 'react';
 import { assign, mapActions, select } from '../util';
 
-const CONTEXT_TYPES = {
-	store: () => {}
-};
+const StoreContext = createContext();
+
+export const Consumer = StoreContext.Consumer;
+
+/** Hook up to the store. Returns state and actions bound to the store. re-renders on change.
+ *  @param {Function|Array|String} mapStateToProps  A function mapping of store state to prop values, or an array/CSV of properties to map.
+ *  @param {Function|Object} [actions] 				Action functions (pure state mappings), or a factory returning them. Every action function gets current state as the first parameter and any other params next
+ *  @param {Object} [props] 				Props from the component. Every action function gets the current state as the first parameter and any other params (such as these props) next
+ *  @returns {Array}
+ *  @example
+ *    const [state] = useStore('foo,bar');
+ *  @example
+ *    const actions = { someAction }
+ *    const [state, boundActions] = useStore('foo,bar', actions)
+ */
+export function useStore(mapStateToProps, actions, props) {
+	props = props || {};
+
+	if (typeof mapStateToProps!=='function') {
+		mapStateToProps = select(mapStateToProps || []);
+	}
+
+	let store = useContext(StoreContext);
+
+	const [state, setState] = useState({});
+
+	function update() {
+		let mapped = mapStateToProps(store ? store.getState() : {}, props);
+		for (let i in mapped) if (mapped[i]!==state[i]) {
+			return setState(mapped);
+		}
+		for (let i in state) if (!(i in mapped)) {
+			return setState(mapped);
+		}
+	}
+
+	useEffect(() =>  store.subscribe(update), []);
+
+	useEffect(() => update(), [props]);
```

Comment:
@DonnieWest missed unsubscribe when unmount component

```suggestion
useEffect(() => {
  update()

  return () => store.unsubscribe()
}, [props])
```

Reply:
Implicit returns are weird, but that should be what I'm doing on line 40 - `subscribe` returns a function that unsubscribes. That function should be used when the component unmounts because it's being returned from useEffect, right?

Reply:
I just messed around with this locally - explicitly returning the unsubscribe function as listed here doesn't seem to make the tests pass.

The bizarre thing though is that slapping a `console.log` _shows_ it's being called